### PR TITLE
Add `mc-sgx-dcap-quoteverify::PathInitializer`

### PR DESCRIPTION
**Pull request by nick-mobilecoin**
Previously the quote verification enclave and quote provider library
paths for `mc-sgx-dcap-quoteverify` were set via
`mc-sgx-dcap-quoteverify::set_path`.  Now there is a one time entry
point of `mc-sgx-dcap-quoteverify::PathInitiliazer` to set the paths.

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #158** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/158" target="_blank">Graphite</a> 
    * **PR #159** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/159" target="_blank">Graphite</a> 
      * **PR #163** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/163" target="_blank">Graphite</a> 
        * **PR #164** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/164" target="_blank">Graphite</a> 
          * **PR #165** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/165" target="_blank">Graphite</a> 
            * **PR #166** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/166" target="_blank">Graphite</a> 
              * **PR #171** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/171" target="_blank">Graphite</a> 
                * **PR #173** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/173" target="_blank">Graphite</a> 
                  * **PR #174** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/174" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/174?utm_source=stack-comment).

**Review by jcape — APPROVED** [review]

File: dcap/quoteverify/src/lib.rs [hunk 1]
```diff
@@ -6,5 +6,30 @@
...
+
+/// Errors interacting with quote verification library functions
+#[derive(Clone, Debug, displaydoc::Display, Eq, Hash, PartialEq, PartialOrd, Ord)]
+#[non_exhaustive]
+pub enum Error {
+    /// Paths have already been initialized
+    PathsInitialized,
+    /// Error from SGX quoting library function: {0}
+    Sgx(Quote3Error),
+    /// Failed ot convert a path to a string.  Path {0}
```
Comment (samdealy) [inline dcap/quoteverify/src/lib.rs:21]: nit: to

File: dcap/quoteverify/src/quote_enclave.rs [hunk 2]
```diff
@@ -7,69 +7,195 @@
...
-/// * `path` does not point to a file
-/// * `path` is longer than 259 (bytes)
-/// * `path` contains a null (0) byte.
+// Using the value from SGX to validate inputs and provide better error
+// messages.
+// NB: This should be checked once converted to a CString as it's the number of
+//  bytes (+ NULL), not the number of characters.
+const MAX_PATH_LENGTH: usize = 260;
+
+/// A convenience type alias for a `Result` which contains an [`Error`].
```
Comment (samdealy) [inline dcap/quoteverify/src/quote_enclave.rs:23]: nit: [`Result`]

File: dcap/quoteverify/src/quote_enclave.rs [hunk 3]
```diff
@@ -7,69 +7,195 @@
...
+                Self::set_path(PathKind::QuoteProviderLibrary, path)
+            })?;
+            *value = Some(PathInitializer);
+            Ok(())
+        } else {
+            Err(Error::PathsInitialized)
+        }
+    }
+
+    /// Will ensure the paths have been initialized
```
Comment (samdealy) [inline dcap/quoteverify/src/quote_enclave.rs:96]: nit: Ensures

**Review by samdealy — APPROVED** [review]

File: dcap/quoteverify/src/lib.rs (hunk 1 above)
Comment (nick-mobilecoin) [reply dcap/quoteverify/src/lib.rs]: done

File: dcap/quoteverify/src/quote_enclave.rs (hunk 2 above)
Comment (nick-mobilecoin) [reply dcap/quoteverify/src/quote_enclave.rs]: done

**Comment by nick-mobilecoin** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/174?utm_source=stack-comment).


Job Reference: 5plz6eqSsqN8GiebVIMp

**Pull request merged**